### PR TITLE
Butts, nuts, n' tats. (please testmerge or merge & deploy before Halloween ball)

### PR DESCRIPTION
**Pull request by BlueWildrose**
## About The Pull Request

In summary, new genital organ: butts. Ported from Hyperstation. Credit to Quotefox for the one who made the asses (if I believe correctly)

To be specific of what this PR adds:

- Readily available enlarged asses for character customization. Much like dicks, they're covered up when you cover up your groin.
- Butt enlargement chemical: Denbu Tincture. Please note you will need to enable the new "Butt enlargement" preference in Content Preferences for this to be used first.
- Butt enlargement pills has been added to the kinkmate.
- Butt reverting chemicals have been added to the organmegaly hypospray kit.
- Fix to testicles. They finally show themselves again.
- Because the testicles are fixed, the "hidden" testicles have been re-added for those who don't want that. Accessible through testicle shapes.
- Updated boob sprites from Hyper. No new boob sizes added then what we already got. I was just told these were better looking, and I feel that I agree.

_(no, not planning on including the hyper dick sprites. I'd like to run an emojicracy or something first for that.)_

Currently there is no way to make butt adjusting chemicals via chemistry, but I can do that at a later time if there's enough demand. There's only like 5 butt sizes so I didn't consider it a major concern.

I had to do a lot of twisting around to get this to work for Citadel considering Hyper hasn't demodularized genitals or anything like that, so let's hope this is worth it. All considering, about 75-85% of the code is my own, especially since I included options to go back to a smaller ass if you decided you didn't want a big ass for the rest of the shift after downing pills or whatever.

## Why It's Good For The Game

_"Oh no Blue, WHY."_

We do have gamers and this is a fairly gamer server in terms of furry server variety, but we're also a horny server in the end, no matter how much gamers we got here. Relax.

Plus this would be something ideal to add to a certain upcoming event that's mostly gonna be social interaction and slice o' life RP.

## Changelog
:cl:
add: Butts.
add: Butt growth related chemicals. Check out the kinkmate, and check out the hypokit for rectifying genitals.
fix: Testicles are actually rendered again. Hidden testicles are available as well.
/:cl:

### CODE REVIEW

**Comment by NicoDeimos** [comment]
Buttbots

**Comment by BlueWildrose** [comment]
> Buttbots

someone else can add that. I'm leaving the butt memes out for the most part.

Besides, you can probably make a circuit that emulates what a buttbot does anyways.

**Comment by shellspeed1** [comment]
Can confirm butt bott circuit exists. Admins hate them.

**Comment by Sishen1542** [comment]
lets not and say we didnt

**Comment by BlueWildrose** [comment]
By request I've updated the breast sprites as well.

**Comment by LordSaladin** [comment]
Honk honk

**Comment by TheEaterOfBeans** [comment]
butt removal time

**Comment by MalricB** [comment]
this is one step closer to fard

**Comment by BlueWildrose** [comment]
> this is one step closer to fard

Yeah farts aren't getting added, ever. If Citadel reaches that state, I'd bail, but I don't expect that at **all** with this community.

**Comment by BlueWildrose** [comment]
Handled all the bug fixes that I realized after having slept and thought through it. Just needs a maintainer lookthrough, and that said and complete, it's ready for merging.

File: code/modules/arousal/genitals.dm [hunk 1]
```diff
@@ -346,20 +352,30 @@
 	var/organCheck = locate(/obj/item/organ/genital) in internal_organs
 	var/breastCheck = getorganslot(ORGAN_SLOT_BREASTS)
 	var/willyCheck = getorganslot(ORGAN_SLOT_PENIS)
+	var/buttCheck = getorganslot(ORGAN_SLOT_BUTT)
+	var/ballCheck = getorganslot(ORGAN_SLOT_TESTICLES)
 
 	if(organCheck == FALSE)
 		if(ishuman(src) && dna.species.use_skintones)
 			dna.features["genitals_use_skintone"] = TRUE
 		if(src.dna.species.fixed_mut_color)
```
Comment (SandPoot) [inline code/modules/arousal/genitals.dm]:
```suggestion
		if(dna.species.fixed_mut_color)
```
let's take the opportunity to fix this ugly!

Comment (BlueWildrose) [reply code/modules/arousal/genitals.dm]: Done

**Comment by silicons** [comment]
controversial pr #18567982 time

**Comment by BlueWildrose** [comment]
Conflict spotted as things got merged. Resolved.

**Comment by Sishen1542** [comment]
> this is one step closer to fard

im PRing farts if this gets merged tbh

**Comment by The-Sun-In-Splendour** [comment]
Delete this pr

**Comment by keronshb** [comment]
> 
> 
> > this is one step closer to fard
> 
> im PRing farts if this gets merged tbh

Same.  Also adding the component for blowing up when farting on a bible.

This is the path to getting slippery shit added again.

**Comment by BlueWildrose** [comment]
Merge conflict resolved (again)

**Comment by keronshb** [comment]
image

**Comment by BlueWildrose** [comment]
Unless you have some actual productive things to say about the PR, set aside the "totally not mocking and trolling I'm only joking it's all in good fun haha", "this is a slippery slope" and "if those horny players get an inch they'll take a mile" shitposts. 

Several people who don't bother going to github had requested this, so I'm going to fulfill that request.

**Comment by LordSaladin** [comment]
I'm just waiting on ss14 to be finished do I don't have to commit to learning byond

**Comment by timothyteakettle** [comment]
yeah sure considering we have vore sprites, genital sprites, i don't know how ass is going too far lmao

**Comment by Mosley1f** [comment]
Aight Imma take this if nobody wants it :3

**Comment by LordSaladin** [comment]
Honkers x badonkas

**Comment by shellspeed1** [comment]
test merge hell

**Comment by LordSaladin** [comment]
why is this still not merged

**Comment by LeadOnTaste** [comment]
It is... just too much. I can understand anuses at Skyrat, but that...

**Pull request merged**